<!--https://github.com/PacktPublishing/ROS-Robotics-By-Example/blob/master/Chapter_5_code/ros_robotics/urdf/gripper.xacro-->
<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <!-- Gripper -->
    <joint name="left_gripper_joint" type="revolute">
        <parent link="top_link"/>
        <child link="left_gripper"/>
        <origin xyz="0 0 ${height2 - axle_offset}" rpy="0 -1.57 0"/>
        <axis xyz="0 0 -1"/>
        <limit effort="30.0" lower="-0.548" upper="0.0" velocity="0.1"/>
    </joint>

    <link name="left_gripper">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://ros_robotics/meshes/l_finger.dae"/>
            </geometry>
        </visual>

        <collision>
            <geometry>
                <box size="0.1 0.1 0.1"/>
            </geometry>
        </collision>
        <xacro:default_inertial z_value="0" i_value="0.000001" mass="1"/>
    </link>

    <joint name="left_tip_joint" type="fixed">
        <parent link="left_gripper"/>
        <child link="left_tip"/>
    </joint>

    <link name="left_tip">
        <visual>
            <origin rpy="0 0 0" xyz="0.09137 0.00495 0"/>
            <geometry>
                <mesh filename="package://ros_robotics/meshes/l_finger_tip.dae"/>
            </geometry>
        </visual>

        <collision>
            <geometry>
                <box size="0.1 0.1 0.1"/>
            </geometry>
        </collision>
        <xacro:default_inertial z_value="0" i_value="1e-6" mass="1e-5"/>
    </link>

    <joint name="right_gripper_joint" type="revolute">
        <parent link="top_link"/>
        <child link="right_gripper"/>
        <origin xyz="0 0 ${height2 - axle_offset}" rpy="0 -1.57 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="30.0" lower="-0.548" upper="0.0" velocity="0.1"/>
    </joint>

    <link name="right_gripper">
        <visual>
            <origin rpy="3.1415 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://ros_robotics/meshes/l_finger.dae"/>
            </geometry>
        </visual>

        <collision>
            <geometry>
                <box size="0.1 0.1 0.1"/>
            </geometry>
        </collision>
        <xacro:default_inertial z_value="0" i_value="1e-6" mass="1"/>
    </link>

    <joint name="right_tip_joint" type="fixed">
        <parent link="right_gripper"/>
        <child link="right_tip"/>
    </joint>

    <link name="right_tip">
        <visual>
            <origin rpy="-3.1415 0 0" xyz="0.09137 0.00495 0"/>
            <geometry>
                <mesh filename="package://ros_robotics/meshes/l_finger_tip.dae"/>
            </geometry>
        </visual>

        <collision>
            <geometry>
                <box size="0.1 0.1 0.1"/>
            </geometry>
        </collision>
        <xacro:default_inertial z_value="0" i_value="1e-6" mass="1e-5"/>
    </link>

    <transmission name="gripper_transmission1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="left_gripper_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="gripper_motor1">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    <transmission name="gripper_transmission2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="right_gripper_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="gripper_motor2">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

</robot>